### PR TITLE
chore(deps): upgrade packages and move to Node.js 14 LTS

### PR DESCRIPTION
**Pull request by raisedadead**
This PR adds these commits, ideally should land individually.

- chore(deps): upgrade root for patches & minor revs
- chore(deps): upgrade lerna to v4
- chore(deps): upgrade eslint, prettier & related packages
- fix(actions): update workflow matrices & docs 

Additionally, this PR will also bump us up to Node.js 14 LTS ~~and npm 7.x~~.

Edit: We should stay on npm 6.x as long as v7 is not being officially shipped with Node.js LTS. However, if you happen to use v7, you should not be facing issues.

### CODE REVIEW

**Comment by raisedadead** [comment]
There are more things that I can upgrade in the PR, but because that linting commit updates 100+ files I am going to wait for now and upgrade the rest of the deps in subsequent PRs.

**Comment by nhcarrigan** [comment]
> However, if you happen to use v7, you should not be facing issues.

Just to point out - npm v7 changes the structure of the package-lock.json.

If a contributor is using npm v7 and runs `npm install` (instead of `npm ci`) then a new package-lock.json with lockfile v2 will be generated and the contributor will need to ensure that isn't committed.

File: docs/how-to-setup-freecodecamp-locally.md [hunk 1]
```diff
@@ -21,10 +21,10 @@ Some community members also develop on Windows 10 natively with Git for Windows
...
+| Prerequisite                                                                                  | Version | Notes                                                                                                                                                                                 |
+| --------------------------------------------------------------------------------------------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Node.js](http://nodejs.org)                                                                  | `14.x`  | [LTS Schedule](https://github.com/nodejs/Release#release-schedule)                                                                                                                    |
+| npm (comes bundled with Node)                                                                 | `7.x`   | Does not have LTS releases, we use the version bundled with Node LTS                                                                                                                  |
```
Comment (nhcarrigan) [inline docs/how-to-setup-freecodecamp-locally.md:27]: This doesn't seem right.

**Comment by raisedadead** [comment]
> > However, if you happen to use v7, you should not be facing issues.
> 
> Just to point out - npm v7 changes the structure of the package-lock.json.
> 
> If a contributor is using npm v7 and runs `npm install` (instead of `npm ci`) then a new package-lock.json with lockfile v2 will be generated and the contributor will need to ensure that isn't committed.

Yes, you are correct. 

We already upgraded the lockfiles with npm v7 a few days ago, and they are backwards compatible. I think we are in the clear here regardless of the npm version anyone is using. 

That said we should stick to whatever version npm is being shipped with LTS at all times to prevent issues say for instance when we install Node.js automatically in our CI/CD workflows.

File: docs/how-to-setup-freecodecamp-locally.md (hunk 1 above)
Comment (raisedadead) [reply docs/how-to-setup-freecodecamp-locally.md:27]: I should fix this momentarily.

**Comment by nhcarrigan** [comment]
Another question (while I'm waiting for my local production build to complete).

Our Cypress tests can be a bit flaky at times and fail erroneously. I know it wouldn't be a DRY approach, but should we split the 12.x and 14.x workflows into two separate instances? Currently, if I need to re-run a failing Cypress test on a PR, I would need to re-run all four tests at once.

**Comment by raisedadead** [comment]
The cypress tests aren't required to be passing for both merging and deployment, so running all jobs is fine in terms of indirect dependency on them.

Something that we would want to consider is adding stages if needed. Functionally separating the scope of the stages to say: auth, challenges, donate, etc. IDK.

Along with [the continue on error](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error) and stages we would be able to isolate the stage, and I think GitHub will let you run that matrix set independently.

This would mean we would consume more runners though.

--

Unrelated note: We will switch to be on Node.js 14 only in a few weeks from now.

**Review by naomi-lgbt — APPROVED** [review]
Pulled this down locally, poked at a development instance and a production-built instance.

I didn't see any issues or concerns. Was able to do things like complete challenges, submit projects, claim certifications, and change settings.

Think this is good to go.

**Comment by raisedadead** [comment]
Just to clarify my previous comments about stages:

A matrix set is every combination of Node.js + browser on a particular stage. So, think of running the auth stage that consists of all the auth tests in a suite and run them on all variants of Node.js and browsers.

We would need to need rejig the suites to be more "scoped" but that would be one way to isolate concerns.

**Comment by nhcarrigan** [comment]
That could be a good approach. Worth diving in to as time allows, certainly.

**Pull request merged**